File: UCF HSPT Documents/2012/Solutions/avengers.c
#include <stdio.h>
int main()
{
    int villains;

    //max size for a villain is 40 characters, but we need the 41st to account for the null character
    char villain[41];
    int i,v;
    freopen("avengers.in", "r", stdin);
    scanf("%d",&villains);

    //Call gets before you begin reading because after we read in the number of villains,
    //the pointer hasn't found a new line character yet. If we had left this out, the
    //first call to gets would only put a blank line in villain because it read up until
    //a new line character (which was the next character from the pointer).
    gets(villain);
    for(v=0;v<villains;v++)
    {
        //We always reset the characters in villain to null just in case the next
        //villain has a name of different length
        for(i=0;i<40;i++)
            villain[i]=0;

        //Use gets(char[]) when you need to read an entire line in C
        gets(villain);

        //We are at the beginning of a third
        if(v%3==0)
        {
            printf("Captain America: I'll take care of %s!\n",villain);
        }
        //We are at the middle of a third
        else if(v%3==1)
        {
            printf("Thor: I shall deal with %s!\n",villain);
        }
        //We are at the end of a third
        else
        {
            printf("HULK SMASH!\n");
        }
        //Don't forget the extra line between villains!
        printf("\n");
    }
    return 0;
}

File: UCF HSPT Team/src/y2012/avengers.java
import java.util.Scanner;
import java.io.File;
import java.io.IOException;


public class avengers {
	public static void main(String[] args) throws IOException
	{
		Scanner br=new Scanner(new File("avengers.in"));
		int villains=br.nextInt();
		//When you use Scanner, after you read in a token (instead of a line), it leaves
		//its reader right after the last token. Then when you read an entire line, it
		//only goes to the new line character, which is usually where your reader is.
		//So we just skip that by calling nextLine() before we actually want to read.
		br.nextLine();
		for(int i=0;i<villains;i++)
		{
			String villain=br.nextLine();
			//If we have started a new third
			if(i%3==0)
				System.out.println("Captain America: I'll take care of "+villain+"!\n");
			//If we are in the middle of a third
			else if(i%3==1)
				System.out.println("Thor: I shall deal with "+villain+"!\n");
			//If we are at the end of a third
			else
				System.out.println("HULK SMASH!\n");
		}
	}
}
